### ISSUE TITLE
fcswatch tool fails with DirectoryNotFoundException

### ISSUE THREAD

**Issue by NatElkins** [humhei/FCSWatch#19]
I wanted to try out the `fcswatch` global tool on my solution to see if it worked.  We've got a main project that's a Giraffe server and it has a bunch of project dependencies.  When I run the `fcswatch` global tool, it builds just fine and then fails with the following error:
```Build succeeded.

/Users/nat/Projects/shopmasse-backend/fsharp/src/Masse.API/RouteHandler.fs(28,48): warning FS0044: This construct is deprecated [/Users/nat/Projects/shopmasse-backend/fsharp/src/Masse.API/Masse.API.fsproj]
/Users/nat/Projects/shopmasse-backend/fsharp/src/Masse.API/RouteHandler.fs(32,55): warning FS0044: This construct is deprecated [/Users/nat/Projects/shopmasse-backend/fsharp/src/Masse.API/Masse.API.fsproj]
    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:13.95
ERRORS: System.IO.DirectoryNotFoundException: Could not find a part of the path '/Users/nat/Projects/shopmasse-backend/fsharp/Masse.Catalog.Sql/Masse.Catalog.Sql.fsproj'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at System.Xml.XmlDownloadManager.GetStream(Uri uri, ICredentials credentials, IWebProxy proxy, RequestCachePolicy cachePolicy)
   at System.Xml.XmlUrlResolver.GetEntity(Uri absoluteUri, String role, Type ofObjectToReturn)
   at System.Xml.XmlTextReaderImpl.OpenUrl()
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.XmlLoader.Load(XmlDocument doc, XmlReader reader, Boolean preserveWhitespace)
   at System.Xml.XmlDocument.Load(XmlReader reader)
   at System.Xml.XmlDocument.Load(String filename)
   at FcsWatch.Core.Types.FullCrackedFsprojModule.loop@202.Invoke(String projectFile) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 209
   at FcsWatch.Core.Types.FullCrackedFsprojModule.loop@202.Invoke(String projectFile) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 215
   at FcsWatch.Core.Types.FullCrackedFsprojModule.loop@202.Invoke(String projectFile) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 215
   at FcsWatch.Core.Types.FullCrackedFsprojModule.loop@202.Invoke(String projectFile) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 215
   at FcsWatch.Core.Types.FullCrackedFsprojModule.loop@202.Invoke(String projectFile) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 215
   at FcsWatch.Core.Types.FullCrackedFsprojModule.loop@202.Invoke(String projectFile) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 215
   at FcsWatch.Core.Types.FullCrackedFsprojModule.easyGetAllProjPaths(String entryProjectFile) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 217
   at FcsWatch.Core.Types.FullCrackedFsprojModule.getAllCrackedFsprojs(String projectFile) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 256
   at FcsWatch.Core.Types.FullCrackedFsprojModule.create@310-7.Invoke(Unit unitVar) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 342
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2)
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit()
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronouslyInCurrentThread[a](CancellationToken cancellationToken, FSharpAsync`1 computation)
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout)
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken)
   at FcsWatch.Core.Types.crackedFsprojBundle@530.Invoke(FSharpMailboxProcessor`1 inbox) in C:\projects\fcswatch\src\FcsWatch.Core\Types.fs:line 547
```

I see it's trying to access a path `'/Users/nat/Projects/shopmasse-backend/fsharp/Masse.Catalog.Sql/Masse.Catalog.Sql.fsproj'` which doesn't exist.  The actual path is `'/Users/nat/Projects/shopmasse-backend/fsharp/src/Masse.Catalog.Sql/Masse.Catalog.Sql.fsproj'`.  I'm not sure why the `src` is getting cut out.

Any idea how I can troubleshoot this issue?  I'm using macOS and `fcswatch` version 0.7.2.

**Comment by NatElkins** [comment]
Another piece of information that might be useful: `Masse.Catalog.Sql` is a transitive project dependency, not a direct one.

**Comment by humhei** [comment]
Hello, @NatElkins 
Can you upload a sample.zip to me so i can reproduce it....
You don't need include the source files,
Only the projects is enough.


Thanks very much for your report

**Comment by NatElkins** [comment]
Hi @humhei ,

Thanks so much for the quick response!

Here are the projects without source code:

[repro-projects.zip](https://github.com/humhei/FCSWatch/files/3090269/repro-projects.zip)

The project I was trying to run is Masse.API.fsproj.

If anyone else wants to easily get the projects for their repo, here's the command I used:

`find . -name '*.fsproj' -exec cp --parents \{\} ~/repro-projects \;` 

`~/repro-projects` needs to be created in advance.  On macOS you'll need to install GNU coreutils, since default `cp` doesn't allow the `--parents` option.

**Comment by humhei** [comment]
Released as 0.7.4
